bioconda_utils/bioconductor_skeleton.py

```diff
@@ -813,8 +812,11 @@ def write_recipe(package, recipe_dir, config, force=False, bioc_version=None,
 
             # Install and clean up
             R CMD INSTALL --library=$PREFIX/lib/R/library --build $TARBALL
-            rm $TARBALL""")
+            rm -r $STAGING""")
```

Comment:
Can we always be sure `$STAGING` has no other content than `$TARBALL`? If not, I'd rather use
```bash
rm $TARBALL
rmdir $STAGING
```

Reply:
`$STAGING` was created at the beginning of this script specifically to store the tarball. To my knowledge other conda processes will not use `$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM`. But even if they did, these files would no longer be needed if the package is removed.

If we do want to allow the possibility of other files to exist there after the package is removed, we'd want to do:

```
rm $TARBALL
rmdir --ignore-fail-on-non-empty $STAGING
```

Otherwise the script would fail due to the directory not being empty.

Reply:
> `$STAGING` was created at the beginning of this script specifically to store the tarball. To my knowledge other conda processes will not use `$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM`. But even if they did, these files would no longer be needed if the package is removed.

I agree, but who knows what people might do ¯\\_(ツ)\_/¯. I just like being explicit when deleting stuff -- but I'm fine either way as that's only being pedantic 😉.

> Otherwise the script would fail due to the directory not being empty.

No, the `post-link.sh` is not executed with the `-e` switch, hence it will not exit on error. Otherwise we'd also have to take care of the previous commands (`mkdir`, `wget`, `rm`) which can also fail any time. IMHO, we should even explicitly not use `--ignore-fail-on-non-empty` to not suppress error output.

Reply:
> I agree, but who knows what people might do ¯\_(ツ)_/¯. I just like being explicit when deleting stuff -- but I'm fine either way as that's only being pedantic

I actually like the 2-step process. It was what I was going to do originally, but feared it was too verbose. I'm happy to switch to using `rmdir`.

> No, the post-link.sh is not executed with the -e switch, hence it will not exit on error.

Good point! I always forget that Bash doesn't stop on error by default.